src/encoding.ts

```diff
@@ -58,9 +58,22 @@ export function detectEncoding(buffer: Buffer): string | null {
     false
   );
   if (experimental) {
-    const detected = chardet.detect(buffer);
-    if (detected) {
-      return detected.replace(/[^a-zA-Z0-9]/g, "").toLocaleLowerCase();
+    const detected = chardet.detectAll(buffer);
+    const encodingPriorities = configuration.get<string[]>(
+      "experimental.encoding_priority",
+      []
+    );
+
+    if (!detected) {
+      return null;
+    }
+
+    for (const pri of encodingPriorities) {
+      for (const det of detected) {
+        if (pri === det.name) {
```

Comment:
Use insensitive case compare, for instance `pri.toLocaleLowerCase() === det.name.toLocaleLowerCase()`

Maybe, you can create a function to fix name, like `fixEncodeName` and return `name.replace(/[^a-zA-Z0-9]/g, "").toLocaleLowerCase()`

Reply:
yeah, good idea